tensorboard/plugins/graph/tf_graph_controls/tf-graph-controls.ts

```diff
@@ -12,397 +12,546 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-(function() { // Private scope.
-  /**
-   * Display only devices matching one of the following regex.
-   */
-  var DEVICE_NAMES_INCLUDE = [
-    {
-      // Don't include GPU stream, memcpy, etc. devices
-      regex: /device:[^:]+:[0-9]+$/,
+namespace tf.graph.controls {
+
+interface DeviceNameExclude {
+  regex: RegExp,
+}
+
+const DEVICE_NAME_REGEX = /device:([^:]+:[0-9]+)$/;
+
+/**
+ * Display only devices matching one of the following regex.
+ */
+const DEVICE_NAMES_INCLUDE: DeviceNameExclude[] = [
+  {
+    // Don't include GPU stream, memcpy, etc. devices
+    regex: DEVICE_NAME_REGEX,
+  }
+];
+
+interface StatsDefaultOff {
+  regex: RegExp,
+  msg: string, // 'Excluded by default since...'
+}
+
+/**
+ * Stats from device names that match these regexes will be disabled by default.
+ * The user can still turn on a device by selecting the checkbox in the device list.
+ */
+const DEVICE_STATS_DEFAULT_OFF: StatsDefaultOff[] = [];
+
+export interface Selection {
+  run: string,
+  tag: string|null,
+  type: tf.graph.SelectionType,
+}
+
+export interface DeviceForStats {
+  [key: string]: boolean;
+}
+
+// TODO(stephanwlee): Move this to tf-graph-dashboard
+export interface TagItem {
+  tag: string|null,
+  displayName: string,
+  conceptualGraph: boolean,
+  opGraph: boolean,
+  profile: boolean,
+}
+
+// TODO(stephanwlee): Move this to tf-graph-dashboard
+export interface RunItem {
+  name: string,
+  tags: TagItem[],
+}
+
+// TODO(stephanwlee): Move this to tf-graph-dashboard
+export interface Dataset extends Array<RunItem> {
```

Comment:
I don’t understand; what’s the point of this interface? If this is just
meant to be a type alias, surely `export type Dataset = Array<RunItem>`
is clearer? Do we actually create classes that extend `Array`?